File: frontend/src/types/feedback.ts
export interface FeedbackResponse {
  id: number;
  email_id: number;
  old_category: string;
  new_category: string;
  reason: string | null;
  created_at: string;
}

export interface FeedbackListResponse {
  items: FeedbackResponse[];
  total: number;
  page: number;
  page_size: number;
}

File: backend/app/schemas/feedback.py
from __future__ import annotations
import datetime
from typing import Optional
from pydantic import BaseModel


class FeedbackResponse(BaseModel):
    id: int
    email_id: int
    old_category: str
    new_category: str
    reason: Optional[str] = None
    created_at: datetime.datetime

    model_config = {"from_attributes": True}


class FeedbackCreateRequest(BaseModel):
    email_id: int
    old_category: str
    new_category: str
    reason: Optional[str] = None
